src/test/fuzz/versionbits.cpp

```diff
@@ -0,0 +1,343 @@
+// Copyright (c) 2020-2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <chain.h>
+#include <chainparams.h>
+#include <consensus/params.h>
+#include <primitives/block.h>
+#include <versionbits.h>
+
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+
+#include <cstdint>
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace {
+class TestConditionChecker : public AbstractThresholdConditionChecker
+{
+private:
+    mutable ThresholdConditionCache m_cache;
+    const Consensus::Params dummy_params{};
+
+public:
+    const int64_t m_begin = 0;
+    const int64_t m_end = 0;
+    const int m_period = 0;
+    const int m_threshold = 0;
+    const int m_bit = 0;
+
+    TestConditionChecker(int64_t begin, int64_t end, int period, int threshold, int bit) : m_begin{begin}, m_end{end}, m_period{period}, m_threshold{threshold}, m_bit{bit}
```

Comment:
style-nit: (add new line and clang-format)

```suggestion
    TestConditionChecker(int64_t begin, int64_t end, int period, int threshold, int bit)
    : m_begin{begin}, m_end{end}, m_period{period}, m_threshold{threshold}, m_bit{bit}
```